event-sourcing/event-sourcing-core/src/main/scala/org/apache/james/eventsourcing/CommandDispatcher.scala

```diff
@@ -45,9 +49,19 @@ object CommandDispatcher {
 class CommandDispatcher @Inject()(eventBus: EventBus, handlers: Set[CommandHandler[_ <: Command]]) {
   Preconditions.checkArgument(hasOnlyOneHandlerByCommand(handlers), CommandDispatcher.ONLY_ONE_HANDLER_PRECONDITION)
 
-  def dispatch(c: Command): Unit = {
-    trySeveralTimes(() => tryDispatch(c))
-      .getOrElse(() => throw CommandDispatcher.TooManyRetries(c, CommandDispatcher.MAX_RETRY))
+  def dispatch(c: Command): Publisher[Unit] = {
+    tryDispatch(c)
+      .filter(worked => worked)
+      .single()
+      .retry(CommandDispatcher.MAX_RETRY, {
+        case _: NoSuchElementException => true
+        case _ => false
+      })
+      .onErrorMap({
+        case _: NoSuchElementException => CommandDispatcher.TooManyRetries(c, CommandDispatcher.MAX_RETRY)
```

Comment:
Normally, reactor throws a `RuntimeException("retries exhausted")` after run out of retries. If that's true, then this catch is useless. Can you check it?